### PR TITLE
Add rudimentary parser for Git config files

### PR DESCRIPTION
**Pull request by OlivierNicole**
Parsing `.git/config` should be useful to get the remote URLs, and hence the project name in Gitlab (the project name, or its ID, is needed to perform project-related requests to the API).

Is it a nice parser? No. Does it already exist elsewhere in a better version? Probably, although I couldn't find it. Git config files seem to be a form of [INI files](https://en.wikipedia.org/wiki/INI_file).

For now, nested sections are not supported properly. (They will parse, but the result will not be a hierarchy of sections.)

### CODE REVIEW

File: cli/gitconfig_types.mli [hunk 1]
```diff
@@ -0,0 +1,5 @@
+type binding = string * string  (* Key/value pair *)
+
+type section = string * binding list (* Section name and contents *)
+
+type t = section list
```
Comment (maiste) [inline cli/gitconfig_types.mli:5]:
```suggestion
type sections = section list
```
Or `config`? I feel like `t` might not be clear for the reader...

Comment (maiste) [inline cli/gitconfig_types.mli:5]:
```suggestion
type sections = section list
```
Or `config`? I feel like `t` might not be clear for the reader...

File: cli/test_gitconfig.ml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+let print_config fmt config =
```
Comment (maiste) [inline cli/test_gitconfig.ml:1]: Shouldn't this file live in `test` instead of `cli`?

Comment (maiste) [inline cli/test_gitconfig.ml:1]: Shouldn't this file live in `test` instead of `cli`?

File: cli/gitconfig_types.mli (hunk 1 above)
Comment (OlivierNicole) [reply cli/gitconfig_types.mli:5]: Sure, `config` seems nice.

File: cli/test_gitconfig.ml (hunk 2 above)
Comment (OlivierNicole) [reply cli/test_gitconfig.ml:1]: Yeah, this was an ugly. Let me move it there.

**Comment by OlivierNicole** [comment]
I implemented @maiste's suggestions, and implemented utility functions to find `.git/config` and read the project name from it.

**Comment by OlivierNicole** [comment]
Things are probably still a bit brittle, there may be corner cases in the parser or in the project name extractor that I didn't consider, but I think it can be a decent start.

**Comment by tmcgilchrist** [comment]
Thanks @OlivierNicole I'll merge this and people can create PRs for the various commands that should use this to *guess* the right project to use.

**Pull request merged**